google/cloud/bigquery/query.py

```diff
@@ -329,18 +333,41 @@ class ScalarQueryParameter(_AbstractQueryParameter):
             Parameter name, used via ``@foo`` syntax.  If None, the
             parameter can only be addressed via position (``?``).
 
-        type_ (str):
-            Name of parameter type.  One of 'STRING', 'INT64',
-            'FLOAT64', 'NUMERIC', 'BIGNUMERIC', 'BOOL', 'TIMESTAMP', 'DATETIME', or
-            'DATE'.
+        type_ (Union[str, google.cloud.bigquery.query.ScalarQueryParameterType]):
+            Name of parameter type. See
+            :class:`google.cloud.bigquery.enums.SqlTypeNames` and
+            :class:`google.cloud.bigquery.enums.SqlParameterScalarTypes` for
+            supported types.
 
-        value (Union[str, int, float, decimal.Decimal, bool, datetime.datetime, datetime.date]):
+        value (Union[ \
```

Comment:
Why the backslashes?

Reply:
IIRC, Sphinx (or something else in the docs build) emits a warning about the badly formatted docstring, thus the param type need to be in a single line.

However, by introducing native type hints, we can start getting rid of duplicating the same info in the docstrings.

Reply:
> However, by introducing native type hints, we can start getting rid of duplicating the same info in the docstrings.

True. I can see how it renders in the docs without these.